git/git.go

```diff
@@ -714,6 +714,63 @@ func CloneWithoutFilters(args []string) error {
 	return nil
 }
 
+// RemoteRefs returns a list of branches & tags for a remote
+// If retrieveFromRemote is true, this information is built by actually calling
+// the remote using 'git ls-remote'. Otherwise the cached remote branch
+// information held locally is used instead.
+func RemoteRefs(remoteName string, retrieveFromRemote bool) ([]*Ref, error) {
+
```

Comment:
Compromise, I'm keeping the `if match := r.FindStringSubmatch(scanner.Text()); match != nil {` because that's such a standard golang idiom. The HEAD test is now a continue.